tests/unittests/Accelerate/vImageTest.mm

```diff
@@ -585,8 +812,33 @@ static void vImageFillBufferWithRandomDataF(vImage_Buffer* buffer) {
         rowPtr += rgbDest.rowBytes;
     }
 
-    vImageBufferFree(&planarRed);
-    vImageBufferFree(&planarGreen);
-    vImageBufferFree(&planarBlue);
-    vImageBufferFree(&rgbDest);
+    vImageTestBufferFree(&planarRed);
+    vImageTestBufferFree(&planarGreen);
+    vImageTestBufferFree(&planarBlue);
+    vImageTestBufferFree(&rgbDest);
+}
+
+TEST(Accelerate, AlphaUnpremultiply) {
+    SetCACompositor(new NullCompositor);
+
+    char fullPath[_MAX_PATH];
+    GetModuleFileNameA(NULL, fullPath, _MAX_PATH);
+    char* executablePath = strrchr(fullPath, '\\');
+    const char* relativePathToPhoto = "\\Photo2.jpg";
+    strncpy(executablePath, relativePathToPhoto, strlen(relativePathToPhoto) + 1);
+    UIImage* photo = [UIImage imageNamed:[NSString stringWithCString:fullPath]];
+
+    vImage_Buffer unpremultipliedBufferSimd, unpremultipliedBufferNormal;
+    const uint8_t alphaVal = 0x80;
+
+    _vImageSetSimdOptmizationsState(false);
+    vImageTestSetAlphaAndUnpremultiply(photo.CGImage, &unpremultipliedBufferNormal, alphaVal);
+
+    _vImageSetSimdOptmizationsState(true);
+    vImageTestSetAlphaAndUnpremultiply(photo.CGImage, &unpremultipliedBufferSimd, alphaVal);
+
+    vImageTestCompare8888Buffers(&unpremultipliedBufferSimd, &unpremultipliedBufferNormal);
```

Comment:
>vImageTestCompare8888Buffers [](start = 4, length = 28)

This should really be a macro, or you wont have the line number information when a specific test fails.